src/DotNetProjectFile.Analyzers/MsBuild/PragmaWarning.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.CodeAnalysis.Text;
+using System.Text.RegularExpressions;
+using System.Xml;
+
+namespace DotNetProjectFile.MsBuild;
+
+/// <summary>Represents a #pragma warning (disable/restore) in a MS Build project file.</summary>
+public readonly struct PragmaWarning(string diagnosticId, bool disable, Location location)
+{
+    /// <summary>The diagnostic ID.</summary>
+    public string DiagnosticId { get; } = diagnosticId;
+
+    /// <summary>Indicates if the diagnostic is disabled (or restored).</summary>
+    public bool IsDisabled { get; } = disable;
+
+    /// <summary>The location of the #pragma warning.</summary>
+    public Location Location { get; } = location;
+
+    /// <inheritdoc />
+    public override string ToString()
+        => IsDisabled
+        ? $"#pragma warning disable {DiagnosticId}@{Location.GetLineSpan().StartLinePosition}"
+        : $"#pragma warning restore {DiagnosticId}@{Location.GetLineSpan().StartLinePosition}";
+
+    /// <summary>Creates a new #pragma warning from an <see cref="XComment"/>.</summary>
+    public static PragmaWarning? New(XComment comment, MsBuildProject project)
+    {
+        var location = Location.Create(project.Path.ToString(), project.Text.TextSpan(comment.LinePositionSpan()), comment.LinePositionSpan());
+        return TryParse(comment.Value, location);
+    }
+
+    /// <summary>Tries to parse a #pragma warning.</summary>
+    public static PragmaWarning? TryParse(string? str, Location location)
+        => str is { }
+        && Pattern.Match(str) is { Success: true } match
+            ? new(
+                match.Groups["DiagnosticId"].Value,
+                match.Groups["Disable"].Value == "disable",
+                location)
+            : null;
+
+    private static readonly Regex Pattern = new(
+        @"^\s*#pragma +warning +(?<Disable>disable|restore) +(?<DiagnosticId>\w+)(\s|$)",
```

Comment:
`enable` is also a valid, but discouraged, alternative to `restore` since C#8 as a left-over of nullability analysis https://github.com/dotnet/csharplang/issues/3986

Reply:
In this context, `enable` would not work, at least not as would expect it to work: a rule that has been disabled but is enabled from a certain point and on.